storm-server/src/main/java/org/apache/storm/daemon/supervisor/timer/SupervisorHeartbeat.java

```diff
@@ -87,12 +153,16 @@ private SupervisorInfo buildSupervisorInfo(Map<String, Object> conf, Supervisor
             ret.put(stringNumberEntry.getKey(), stringNumberEntry.getValue().doubleValue());
         }
 
+        LOG.debug(NormalizedResources.RESOURCE_NAME_NORMALIZER.normalizedResourceMap(ret).toString());
         return NormalizedResources.RESOURCE_NAME_NORMALIZER.normalizedResourceMap(ret);
     }
 
     @Override
     public void run() {
-        SupervisorInfo supervisorInfo = buildSupervisorInfo(conf, supervisor);
-        stormClusterState.supervisorHeartbeat(supervisorId, supervisorInfo);
+        Map<String, Object> validatedNumaMap = Utils.getNumaMap(conf);
```

Comment:
I think line should be pulled into buildSupervisorInfo(). It appears that the intent is to create as many supervisorInfo records as there are Config.SUPERVISOR_NUMA_META entries with any leftover creating another SupervisorInfo(). And private buildSupervisorInfo() method is never called with validatedNumaMap of null. Each of these supervisorInfos creating a heartbeat-er.

Reply:
The supervisor info does one thing - it creates an object with information about the actual supervisor. I think it's better to not add something like the NUMA heartbeating to its purview.